Angular InstantSearch/routing-seo-friendly/src/app/search/search.component.ts

```diff
@@ -0,0 +1,120 @@
+import { Component } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
+import { history as historyRouter } from "instantsearch.js/es/lib/routers";
+import algoliasearch from "algoliasearch/lite";
+
+// Returns a slug from the category name.
+// Spaces are replaced by "+" to make
+// the URL easier to read and other
+// characters are encoded.
+function getCategorySlug(name) {
+  return name
+    .split(" ")
+    .map(encodeURIComponent)
+    .join("+");
+}
+
+// Returns a name from the category slug.
+// The "+" are replaced by spaces and other
+// characters are decoded.
+function getCategoryName(slug) {
+  return slug
+    .split("+")
+    .map(decodeURIComponent)
+    .join(" ");
+}
+
+const searchClient = algoliasearch(
+  "B1G2GM9NG0",
+  "aadef574be1f9252bb48d4ea09b5cfe5"
+);
+
+@Component({
+  selector: "app-search",
+  templateUrl: "./search.component.html"
+})
+export class SearchComponent {
+  constructor(private route: ActivatedRoute) {}
+  config = {
+    indexName: "demo_ecommerce",
+    searchClient,
+    routing: {
+      router: historyRouter({
+        windowTitle({ category, query }) {
+          const queryTitle = query ? `Results for "${query}"` : "Search";
+
+          if (category) {
+            return `${category} – ${queryTitle}`;
+          }
+
+          return queryTitle;
+        },
+
+        createURL({ qsModule, routeState }) {
+          const categoryPath = routeState.category
+            ? `${getCategorySlug(routeState.category)}/`
+            : "";
+          const queryParameters = {} as any;
+
+          if (routeState.query) {
+            queryParameters.query = encodeURIComponent(routeState.query);
+          }
+          if (routeState.page !== 1) {
+            queryParameters.page = routeState.page;
+          }
+          if (routeState.brands) {
+            queryParameters.brands = routeState.brands.map(encodeURIComponent);
+          }
+
+          const queryString = qsModule.stringify(queryParameters, {
+            addQueryPrefix: true,
+            arrayFormat: "repeat"
+          });
+
+          return `/search/${categoryPath}${queryString}`;
```

Comment:
This likely won't work with a custom pathname.

E.g.: `https://angular-instantsearch.netlify.com/examples/e-commerce/`

It will override `examples/e-commerce`.

We need to prepend the original pathname. Something like this:

```js
const urlParts = location.pathname.match(/^(.*?)\/search/);
const originalPathName = urlParts ? urlParts[1] : '';
```

Reply:
I understand we need that for the demos we published on `https://community.algolia.com/angular-instantsearch/examples/*` but that's rather a details related to the way we publish our examples.

This example runs on `http://localhost:<port>/search` and this implementation of createURL simple and completely sufficient.

Reply:
Go ahead then!

FYI you can see the issue when deployed on algolia/instantsearch.js#3921.